app/controllers/ChargeSummaryController.scala

```diff
@@ -222,6 +223,27 @@ class ChargeSummaryController @Inject()(val auth: AuthenticatorPredicate,
     }
   }
 
+  private def checkForPoaExtraChargeLink(chargeDetailsForTaxYear: FinancialDetailsModel, documentDetailWithDueDate: DocumentDetailWithDueDate, isAgent: Boolean)(implicit user: MtdItUser[_]): Option[String] = {
+    val chargeItem: Option[ChargeItem] = ChargeItem.tryGetChargeItem(isEnabled(CodingOut), isEnabled(ReviewAndReconcilePoa))(
+      chargeDetailsForTaxYear.financialDetails)(documentDetailWithDueDate.documentDetail)
+
+    val desiredMainTransaction: String = chargeItem match {
```

Comment:
worth to move this code as a partial implementation
under ChargeType, as opposite to ChargeType.fromCode function;
something like fromChargeType etc